### PR TITLE
Controls: Add parameter to control visibility of control groups

### LINKED ISSUE #16284: Give more fine tuning control to the existing Controls (auto) discoverability
**Is your feature request related to a problem? Please describe**

Disclaimer - this might be already possible (how?) - I did not find a way from what I tried until now.

I currently work with Angular but I believe this capability is applicable for all the frameworks. For the moment the Controls are binded (autodiscovered) using @Input (props in React) and to the existing class members/functions.

This is probably useful in many situations since it is automating the auto-discoverability of all the fields / Controls (probably for creating quick "kitchen sink" kind of component playgrounds.)
 
However components will usually have many @Input(s) or other members or member functions many of each might not be relevant when testing the component with Storybook. Currently if a component has 30 members there will be 30 Controls "autodiscovered" and the current approach for disabling what is not required is to use the argTypes to disable each Control / member individually. So, if I am interested only to have two out of these 30 members I need to provide boilerplate argTypes to disable the 28 which are not required in the Controls area. 
 
**Describe the solution you'd like**

Have some kind of mechanism to include / exclude Controls in bulk. Maybe you could get some inspiration from how "include" and "exclude" sections work in tsconfig.json

**Describe alternatives you've considered**

I understood that I can use argTypes to exclude each Control individually (exclude works also in bulk?).

### PR DESCRIPTION
**Pull request by konstantindenerz**
Closes #16284

In some environments Storybook renders controls grouped by a category like properties, methods and inputs (Angular) and so on. In some cases only a subset of controls is necessary.

## What I did
I added a new optional parameter to control the visibility of control groups. 

``` js
export const parameters = {
  controls: {
    onlyVisibleCategories: ['inputs'],
  },
};
```

### Before
image

### After
image

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [x] Does this need a new example in the kitchen sink apps?
- [x] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by MichaelArestad** [comment]
@konstantindenerz Thank you for making this PR. @shilman, since you recently worked on conditional controls, can you please take a look at this?

File: addons/controls/src/controls-parameters.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { PresetColor, SortType } from '@storybook/components';
+
+export interface ControlsParameters {
+  sort?: SortType;
+  expanded?: boolean;
+  presetColors?: PresetColor[];
+  hideNoControlsWarning?: boolean;
+  onlyVisibleCategories?: string[];
```
Comment (shilman) [inline addons/controls/src/controls-parameters.ts:8]:
```suggestion
  visibleCategories?: string[];
```

**Review by shilman — COMMENTED** [review]
WDYT about this rename @konstantindenerz ?

**Comment by konstantindenerz** [comment]
@shilman Thanks for the review.  I renamed it based on your recommendation.

**Comment by ndelangen** [comment]
@konstantindenerz could you update this PR with next? .. fixing the merge conflicts.

@shilman if this goes green, can it be merged?

**Comment by konstantindenerz** [comment]
I rebased the PR and created new story file in the new structure. 
@ndelangen @shilman

**Comment by ndelangen** [comment]
@konstantindenerz could you give us maintainers the ability to push to your fork?

I'd like to be able to keep this updated with changes..

**Comment by konstantindenerz** [comment]
I changed the settings. @ndelangen

**Comment by ndelangen** [comment]
Thank you, we'll get this merged after 7.0 is released, I think.

**Comment by gvlekke** [comment]
@ndelangen Hi Nobert, is this already release, I could not find it in the documentation and also don't see a merge.
@konstantindenerz did close it but Im not sure why.

I really would want this feature if thats possible :)

**Pull request closed without merging**